src/main/java/com/google/api/codegen/transformer/go/GoSurfaceNamer.java

```diff
@@ -213,4 +213,14 @@ public String getStubName(Interface service) {
   public String getCreateStubFunctionName(Interface service) {
     return getGrpcClientTypeName(service).replace(".", ".New");
   }
+
+  @Override
+  public String getStaticLangStreamingReturnTypeName(Method method, MethodConfig methodConfig) {
+    // Unsafe string manipulation: The name looks like "LibraryService_StreamShelvesClient",
+    // neither camel or underscore.
+    return String.format(
+        "%s_%sClient",
+        converter.getTypeName(method.getParent()).getNickname(),
+        method.getSimpleName());
```

Comment:
I think you should compose the %sClient part using the Name class, and then do the remaining composition with plain concatenation.

Side question: why does the return type name look like it is a client type name?